### PR TITLE
Fix doc warnings

### LINKED ISSUE #1965: Fix readthedocs warnings
When built locally, there are warnings like

```
Cirq/docs/api.rst:72:<autosummary>:1: WARNING: Inline substitution_reference start-string without end-string.
Cirq/docs/api.rst:72:<autosummary>:1: WARNING: Inline substitution_reference start-string without end-string.
Cirq/cirq/circuits/circuit.py:docstring of cirq.Circuit.apply_unitary_effect_to_state:8: WARNING: Unexpected section title or transition.

--------------
Cirq/cirq/circuits/circuit.py:docstring of cirq.Circuit.to_unitary_matrix:7: WARNING: Unexpected section title or transition.

--------------
Cirq/cirq/sim/density_matrix_simulator.py:docstring of cirq.DensityMatrixStepResult.density_matrix:23: WARNING: Field list ends without a blank line; unexpected unindent.
Cirq/cirq/sim/density_matrix_simulator.py:docstring of cirq.DensityMatrixTrialResult:23: WARNING: Field list ends without a blank line; unexpected unindent.
Cirq/cirq/__init__.py:docstring of cirq.ISWAP:1: WARNING: Inline substitution_reference start-string without end-string.
Cirq/cirq/ops/common_gates.py:docstring of cirq.ISwapPowGate:1: WARNING: Inline substitution_reference start-string without end-string.
Cirq/cirq/sim/sparse_simulator.py:docstring of cirq.SparseSimulatorStep.qubit_map:1: WARNING: duplicate object description of cirq.SparseSimulatorStep.qubit_map, other instance in Cirq/docs/generated/cirq.SparseSimulatorStep.rst, use :noindex: for one of them
Cirq/cirq/sim/sparse_simulator.py:docstring of cirq.SparseSimulatorStep.state_vector:19: WARNING: Field list ends without a blank line; unexpected unindent.
Cirq/cirq/sim/wave_function.py:docstring of cirq.StateVectorMixin.qubit_map:1: WARNING: duplicate object description of cirq.StateVectorMixin.qubit_map, other instance in Cirq/docs/generated/cirq.StateVectorMixin.rst, use :noindex: for one of them
Cirq/cirq/sim/wave_function.py:docstring of cirq.StateVectorMixin.state_vector:20: WARNING: Field list ends without a blank line; unexpected unindent.
Cirq/cirq/study/trial_result.py:docstring of cirq.TrialResult.measurements:1: WARNING: duplicate object description of cirq.TrialResult.measurements, other instance in Cirq/docs/generated/cirq.TrialResult.rst, use :noindex: for one of them
Cirq/cirq/sim/wave_function_simulator.py:docstring of cirq.WaveFunctionTrialResult.state_vector:19: WARNING: Field list ends without a blank line; unexpected unindent.
Cirq/cirq/sim/wave_function.py:docstring of cirq.density_matrix_from_state_vector:12: WARNING: Literal block expected; none found.
Cirq/cirq/google/sim/xmon_simulator.py:docstring of cirq.google.XmonStepResult.qubit_map:1: WARNING: duplicate object description of cirq.google.XmonStepResult.qubit_map, other instance in Cirq/docs/generated/cirq.google.XmonStepResult.rst, use :noindex: for one of them
Cirq/cirq/google/sim/xmon_simulator.py:docstring of cirq.google.XmonStepResult.state_vector:19: WARNING: Field list ends without a blank line; unexpected unindent.


```

### PR DESCRIPTION
**Pull request by andbe91**
Fixes #1965 

This fixes most warnings which occurred when building the documentation. There are three warnings left:

>/home/andbe/Cirq/cirq/circuits/circuit.py:docstring of cirq.Circuit.apply_unitary_effect_to_state:8: >WARNING: Unexpected section title or transition.
>/home/andbe/Cirq/cirq/circuits/circuit.py:docstring of cirq.Circuit.to_unitary_matrix:7: WARNING: >Unexpected section title or transition.

These two should be solved when `apply_unitary_effect_to_state` and `to_unitary_matrix` are removed. The warnings are due to the deprecation warnings.

Finally,
>home/andbe/.virtualenvs/cirq37/lib/python3.7/site-packages/sphinx_rtd_theme/search.html:20: RemovedInSphinx30Warning: To modify script_files in the theme is deprecated. Please insert a <script> tag directly in your theme instead.

This will be solved in the next release of `sphinx_rtd_theme`, see [https://github.com/readthedocs/sphinx_rtd_theme/issues/739](https://github.com/readthedocs/sphinx_rtd_theme/issues/739).

### CODE REVIEW

File: cirq/google/sim/xmon_stepper.py [hunk 1]
```diff
@@ -346,7 +346,7 @@ def simulate_measurement(self, index: int) -> bool:
             index: Which qubit is measured.
 
         Returns:
-            True iff the measurement result corresponds to the |1> state.
```
Comment (Strilanc) [inline cirq/google/sim/xmon_stepper.py]: Wait, why was this a warning?

Comment (andbe91) [reply cirq/google/sim/xmon_stepper.py]: Ah sorry, this was not a warning, just a typo that I noticed while going through the code.

File: cirq/sim/wave_function.py [hunk 2]
```diff
@@ -93,13 +94,16 @@ def density_matrix_of(self, qubits: List[ops.Qid] = None) -> np.ndarray:
...
 
         For example:
-            self.state_vector() = np.array([1/np.sqrt(2), 1/np.sqrt(2)],
-                dtype=np.complex64)
-            qubits = None
-            gives us \rho = \begin{bmatrix}
-                                0.5 & 0.5
-                                0.5 & 0.5
-                            \end{bmatrix}
+        self.state_vector() = np.array([1/np.sqrt(2), 1/np.sqrt(2)],
```
Comment (dabacon) [inline cirq/sim/wave_function.py]: can this be indented if there is a newline between "For example" and the code?

Comment (andbe91) [reply cirq/sim/wave_function.py]: Unfortunately no, if I indent this it somehow creates a warning, `WARNING: Field list ends without a blank line; unexpected unindent.`, even though it renders just fine.

File: cirq/study/trial_result.py [hunk 3]
```diff
@@ -76,11 +76,6 @@ class TrialResult:
 
     Attributes:
         params: A ParamResolver of settings used when sampling result.
-        measurements: A dictionary from measurement gate key to measurement
```
Comment (vtomole) [inline cirq/study/trial_result.py]: Why was this removed? Was it moved somewhere else?

File: cirq/sim/wave_function.py (hunk 2 above)
Comment (vtomole) [reply cirq/sim/wave_function.py]: What happens if you put that command in one line?
 ```
For example:
            self.state_vector() = np.array([1/np.sqrt(2), 1/np.sqrt(2)], dtype=np.complex64)
```

File: cirq/study/trial_result.py (hunk 3 above)
Comment (andbe91) [reply cirq/study/trial_result.py]: It is documented in the Args section of `__init__`.
The reason it had to be removed is that it caused a conflict with the property `measurements` defined further down.  This was the reason for `Cirq/cirq/study/trial_result.py:docstring of cirq.TrialResult.measurements:1: WARNING: duplicate object description of cirq.TrialResult.measurements, other instance in Cirq/docs/generated/cirq.TrialResult.rst, use :noindex: for one of them`

Comment (andbe91) [reply cirq/study/trial_result.py]: Perhaps it could or should be documented in the property method instead.

File: cirq/sim/wave_function.py (hunk 2 above)
Comment (andbe91) [reply cirq/sim/wave_function.py]: It renders the same as no indentation.

Comment (vtomole) [reply cirq/sim/wave_function.py]: Does it give a warning?

File: cirq/study/trial_result.py (hunk 3 above)
Comment (vtomole) [reply cirq/study/trial_result.py]: Ah. Yeah, this seems like it's redundant.

File: cirq/google/sim/xmon_stepper.py (hunk 1 above)
Comment (mpharrigan) [reply cirq/google/sim/xmon_stepper.py]: This isn't a typo, `iff` is a mathematical shorthand for "if and only if". By removing the extra f, the docstring no longer precludes the possibility that you could also return `True` if the result corresponds to the |0> state :)

**Comment by mpharrigan** [comment]
awesome

File: cirq/google/sim/xmon_stepper.py (hunk 1 above)
Comment (andbe91) [reply cirq/google/sim/xmon_stepper.py]: Oh, I see... Today I learned, thanks! Reverted back.

**Comment by mpharrigan** [comment]
@vtomole all good?

File: cirq/sim/wave_function.py (hunk 2 above)
Comment (vtomole) [reply cirq/sim/wave_function.py]: @mpharrigan This one ^ left :)

Comment (andbe91) [reply cirq/sim/wave_function.py]: @vtomole No, it doesn't give a warning. However to get any change in the generated doc, it needs to be a newline as suggested by @dabacon. But that causes some interference with the indented math further down, giving rise to the original warning. 

This is how things looked before:
image
image

To which I prefer the new looks:
image
image

It would be nice to get the code examples in their own boxes. What @dabacon  suggested yields:
image
Which looks slightly nicer. However, that generates warnings...

**Comment by dabacon** [comment]
I will get merge bot running

**Comment by dabacon** [comment]
Has a conflict with xmon_simulator now :(
